### ISSUE TITLE
Make the gateway do something

### ISSUE THREAD

**Issue by liltiptop** [d3athrow/vgstation13#9404]
So there's a gateway on every station, and it does absolute shit besides adminbus. What should we do with it? I think having a gateway open to some object near the station, (abandon planet, vox planet, plasmaman planet, abandon station/ship, xeno planet) and have some crew go on a salvage run, or even try to make a colony on there, like dwarf fortress, would add FUN.
Thoughts?

**Comment by ihadtoregisterforthis** [comment]
#3715
#9236

**Comment by Killette2** [comment]
Getting free toys from gateway is a SHIT idea.
Making a base there and colonizing it, doesn't seem that bad. 
Currently, the gateway is just another milf APC.

We could also make gateways a blob-like RTS where we send monkeys and make them do shit.

**Comment by Exxion** [comment]
You know, we do have procedural generation abilities
Maybe it could beam to procedurally-generated dungeons for tonsafun

**Comment by Shadowmech88** [comment]
@Exxion 
I like the idea, but I think it should randomly activate at some point during the round, and not every round.
If it's active every round at round start, everyone's just going to abandon their posts to rush into the dungeon.

**Comment by Exxion** [comment]
There could be a cap on how many people get to go through
Perhaps even a cap on how many people of specific jobs get to go through (like five security, three engineers, five civilian, for example)
And yeah, not at roundstart, as that would just be bad in general

**Comment by FudgePucker** [comment]
Thats debatable.
Goon has the telescience areas available every round as long as you know telescience enough to get to them.

They have guns, chems, more guns, traitor items, even more guns, did I mention unique overkill guns such as the safari rifle that can crit in one hit or the AK47?

But they are for the most part over looked every round, why?
Because they are available every round, people will stop going to them eventually.

So with our gateway that only worked once in a blue moon that had everyone pour into it to go on adventures, we have the thought now that we can't have it on all the time because everyone will abandon the station to go on adventures. The real reason why they abandon the station to go on adventures is because it is only open rarely, do you want to have another boring round or go on a once in a long time ADVENTURE? If it became 'do the same shift' or 'do the same adventure' then less people will ditch work to go on adventures.

**Comment by Probe1** [comment]
When we briefly had away missions available last August people went on them every single round for the week and a half they were in.  There was no sign of people tiring and it constantly brought high end loot to the station.

**Comment by liltiptop** [comment]
@Probe1 Right, I was there for that. It's what I had in mind for "explore areas".
You know, with all these vaults, we could have people make areas for people to explore, like a map maker kind of thing.

**Comment by Probe1** [comment]
We have plenty of vaults and plenty more up for review and adding.  But vaults aren't away missions and are designed to be hard to find in mind.  We can fix away missions and realign the gateway but the fundamental issue is away missions were pretty badly designed.

A lot of vaults people hate (see: syndie vaults and other loot grabs) are the same thing that made no one WANT to fix away missions.  They were all just glorified "Go get some phat lewt 420".

**Comment by liltiptop** [comment]
@Probe1 Ah, I see what you're saying. If we didn't want to do the colony thing on planets, or other abandon ships, then there's really no reason to have it on the maps, as teleporter does it's job for it.

**Comment by PJB3005** [comment]
Procedural away missions are the only way.

**Comment by Probe1** [comment]
ah you just like em and want to do it.  which is fine.

**Comment by Killette2** [comment]
I just mentioned that gateway missions should not be lootgrabs, but have interesting content to them. Also what if we merged the gateway with telescience as a job somehow, and a way to decode said locations once balanced gateway missions have been added?

**Comment by Probe1** [comment]
They should be but they weren't and it's very challenging to make away missions that aren't just loot grabs.  It pretty much involves coding a tremendous amount of new game code where unique or at least novel things can happen.  It's a big PITA

**Comment by Killette2** [comment]
Give me $5 and I'll make at least 1 good gateway away mission. As a bonus I'll make sure to add as much cheese prevention, and take at least an hour in the construction, probably going to take days since I have yet to make a map but it will have Killette2 quality™

**Comment by Exxion** [comment]
I'm not really sure how to make away missions that are interesting but not lootgrabs
Because there needs to be some reason to do them, but
1. It's unlikely our playerbase would be interested in any kind of story thing, and even if they were, that would have to go in the secret repo
2. Exploration isn't a motivator for pre-mapped things, as people will explore them, figure out all the secrets, and be done with them forever, and making procedural generation that yields things that are interesting to explore would be quite a feat

Perhaps we could have rewards similar to random large xenoarch finds, so not only would the missions be different each time, but so would the rewards.

**Comment by Killette2** [comment]
$5 and I can show you the light - also the randomized loot is still loot, with even more incentive because when you don't know what the loot is it may still be a good thing. Compared to now, you know if the loot is good or not depending on the mission.

**Comment by clusterfack** [comment]
Kek charging for code already when you've never even coded something in the first place?

**Comment by nuklearcellphoneg** [comment]
Just delete the gateway. Nothing good will ever come of it.

**Comment by ghost** [comment]
Make activation of the gateway an admin-controlled button

Also, we need some rework on random loot spawns, maybe some tiering to them

**Comment by Probe1** [comment]
If someone did the code to make it admin activated only, I'd fix the maps in an afternoon.

**Comment by Killette2** [comment]
> Making a map
> Pure coding

Sure clusterfack, I'm sure you need to exclusively code with a map editor.

**Comment by Shadowmech88** [comment]
Well, you've never mapped either.

**Comment by Killette2** [comment]
No, I haven't. I would ask the people who would want me to make the map if there's an experienced mapper in the instance I need a question so I don't have to half ass my project. Honestly I don't even have a way to accept $5 online, it was more of the the principal thing. I have general ideas of what would make fruitful away missions without it being a lootgrab, but I just don't have the motivation do map, or even play space in the first place until there's new interesting features.

**Comment by Shadowmech88** [comment]
Just seems a bit entitled to ask for money in exchange for a feature when you haven't even made a PR before.

**Comment by Killette2** [comment]
It was more of a joke pointed at how long we've had these crappy gateway maps. Nobody has put out any good ones over the years the gateway has existed.

However I can also take random things that mean next to nothing as payment, such as DJ for a round, random racial whitelist like unathi. Anything that will be cool.

**Comment by Probe1** [comment]
No one has mapped anything for it because out of the last 20 months, the gateway has been active and working for exactly 1 of them.

If it worked, people would support it.  @Unid15 recognized it wasn't working and went a different route (and in some ways much better) with vaults.  Look at the flood of support they've received.  They have several advantages.
1.  Very small, quick mapping projects.
2.  Rare so quality isn't as high of a priority.
3.  Rare so creativity and crazyiness is a little more tolerable (If we had a Dance Dance Revolution away mission people would be saying WOW RIGHT IN MY EMULSION).
4.  They're supported by vault coding and it works.

Away missions..
1.  Are not supported by the gateway and regardless of your effort players won't get to experience them.

Full stop.

Turn the gateway on, make it ADMIN TOGGLE while we sort out how best to approach it, and I guaran-fuckin-tee the existing away missions will get fixed in an afternoon.  Then we'll see a trickle of new ones since there's no longer a barrier to them being in play.

**Comment by Killette2** [comment]
That doesn't address the issue of them being lootgrabs. Vaults are loot grabs, which are not guaranteed encounters, and have a random chance of even spawning. Nobody complains about rare or almost non existing loot. People only complain when said loot happens far too often.

An example would be if bananium ore was available all over the asteroid. There would be HONK mechs every round, and people bitching about them happening too often and ruining fun because of the power imbalance they have. Since bananium is only available at a somewhat rare place, it is not anything that happens every round, and if some joe shmoe powergames to make it happen every round, they are dealt with by admins.

With away mission lootgrabs, anyone can, and will, grab the loot because it takes no effort and is almost always right there. Nobody is going to steal a space suit for vaults, because there's no promise of easy loot. I'm not saying that away missions should have random loot, I think the lootgrab thing for away missions should be ditched entirely and it should be just as implied, an away mission. Leave in lootgrabbing for rare places like vaults, and make away missions something you do not for loot, but for something else.

**Comment by Probe1** [comment]
No, it doesn't at all.  But we have to go somewhere to get somewhere.  Things are broke, no one wants to fix em.  Gotta fix something.

**Comment by Killette2** [comment]
I can fix gateway maps, you just have to fix something else first

my motivation

![1453613034848](https://cloud.githubusercontent.com/assets/5169310/14511742/a4ddf722-01a7-11e6-9004-5a3f04889012.png)

**Comment by Probe1** [comment]
Remember the last 20 posts?

Fixing the missions is piss easy, all they have are some bad paths.  The real problem is they make it so every single round the crew are loaded up with nuke ops armor, elite ballistic/energy weapons, traitor items, magic spellbooks et cetera.

Now if the rounds boring and whatnot hey fine open it up.  But as it was with just an open gateway that unlocks 30 minutes in, it just floods the round with (relatively) easy to acquire high end PHAT LEWT which causes more chaos than its worth and pretty much disrupts the round.

**Comment by Killette2** [comment]
I don't think you and I are using the word fix with the same context.

You are saying they can be fixed as they are broken and just need to be working, I say fixed as in it's bad content and needs to be reworked in better maps and design, instead if a location with loot laying all around.

**Comment by MercuryDoll** [comment]
What i'd like to see is a randomly generated dungeon with a few pre-made rooms built in. Similar to how the asteroid gets created, or (i hope noone will witchhunt me for this) the caves in Lifeweb, except much more elaborate and filled with more or less hostile mobs. 
Now i haven't been there for the "vaults", but if it's what i think it is, it would be nice to have them as rare chance to spawn in the dungeon, with a puzzle or something to open them depending on what type. That in addition to rooms with rather useless loot like statues and gems.

**Comment by Killette2** [comment]
RNG would be an improvement, but I think a cleverly designed dungeon would be more fun, it doesn't even need monsters it could just be a puzzle.

Now about that $5...

**Comment by icantthinkofanameritenow** [comment]
I have an idea as well. Perhaps if we made the gateway's opening dependent on a set of prerequisites randomly generated at roundstart, the issue of the loot all being available early on is avoided. I was considering science stuff like "have X number of research types maxed out" or  "make a bomb with dev X or higher and test it successfully", but I'm not averse to having other more exotic requirements as well. The important thing is that they would be random and the crew wouldn't even know that they were possible prerequisites until the gateway becomes usable. 

As for the maps, if someone can fix the paths I think I could try making changes to some of them. It's weird that at least a few of the away missions actually have inferior loot to some of the vaults that have been added since then; I can only assume the difficulty of actually finding a vault is the main difference between them and away missions, other than the size they tend to be of course.

**Comment by Killette2** [comment]
I don't know, a lot of away missions give access to guns, space suits, and some I recall even energy swords. But the loot isn't the only issue, missions being only loot grabs with sometimes a few mobs is the issue.

**Comment by icantthinkofanameritenow** [comment]
@Killette2 
I guess I could go and sprinkle in a few more mobs in some of the maps, but even then I get the feeling that may not be enough. I'd like it if there was a way to implement puzzles of some kind in there somewhere that would actually force players to think a bit as well as robusting idiot NPCs. Something like Deff's Goonecode, but less outright lethal and more cerebral, as it were. 

Anyone have an idea on how to pull that off? Because I'm kinda clueless about how to make it work.

**Comment by Shadowmech88** [comment]
I remember there being an emitter maze away mission that was something of a puzzle.

**Comment by gbasood** [comment]
Part of the problem with away missions is the lack of control for admins. Better tools for that is the key.

Could have an away mission that is a diplomatic mission to a local planet or something, Star Trek style. But the admins would need more control over what's going on in that scenario than they would currently.

**Comment by icantthinkofanameritenow** [comment]
@Shadowmech88 
I've heard of it too, but apparently that was when emitters were pulsed on and off in a distinct pattern rather than emitting a constant beam. With the way they work now the maze would be literally impossible to solve without taking massive amounts of Dermaline. Assuming the emitters would even work, that is.

**Comment by Probe1** [comment]
Unfortuantely, its retardedly easy to cheese since emitters are blocked by a number of things from girders to lockers and so on.

Which is exactly what happened when it was made available.  Either you knew the trick or you died in droves.

**Comment by ihadtoregisterforthis** [comment]
@gbasood But at some point adding too much admin control makes adminless experience empty and unventful.
Adminless meaning when no admins are on, when admins with not enough flags are on, and admins that cower to the mere mention of "bus" and or "RP"

**Comment by gbasood** [comment]
@ihadtoregisterforthis Away missions are literally admin-triggered only right now though. For good reason.

We can't properly balance an away mission and it takes away from the game experience if it's automated, because people run off the station to go on loot quests

**Comment by Killette2** [comment]
You just need good puzzle maps with a reward that isn't loot you can take with you why isn't this easy to understand REEEEEE

**Comment by Shadowmech88** [comment]
What kind of reward would people want that isn't loot?

**Comment by Killette2** [comment]
Fine let me share an example I would use in my own maps, which I'd personally like to make myself for quality™ but I'm probably never going to.

A staff of change turret at the end that's set at random, busting the turret will obviously not leave a retrievable staff. However, as long as you make it to the end of the maze, you can let the staff zap you as much as you want. Free changes to any race, but you can't take the staff itself. You have to think outside of the box see what type of ideas I'm getting at?

**Comment by ihadtoregisterforthis** [comment]
![but](https://cloud.githubusercontent.com/assets/13305885/14547463/47a0066c-0284-11e6-8ea8-9085e0d4c5e4.png)

**Comment by icantthinkofanameritenow** [comment]
@Probe1 
Anything you might suggest to take the place of them as puzzle elements?

**Comment by icantthinkofanameritenow** [comment]
@Killette2 
Most people wouldn't see that as a reward at all, seeing as they choose the race they want to play at roundstart. Unless they want to play as catbeasts or something of their own will, in which case they should be immediately banned for the good of the server. (Yes, I know, skeletons. But why jump through all those hoops when a single SE mutation can accomplish the same thing with far less trouble?)

**Comment by Shadowmech88** [comment]
@icantthinkofanameritenow 
What about mutons, slimes, and xenomorphs?

**Comment by icantthinkofanameritenow** [comment]
@Shadowmech88 

> wanting to play as mutons

And I doubt that the badmins would appreciate something that has a chance of turning people into antags being that accessible. Point is, away missions need concrete rewards or nobody will go for them.

**Comment by gbasood** [comment]
> You just need good puzzle maps with a reward that isn't loot you can take with you why isn't this easy to understand REEEEEE

Once the solution is found everyone will instantly solve it???

**Comment by icantthinkofanameritenow** [comment]
@gbasood 
Good point. Plus, loot you can't bring back isn't even loot.

**Comment by Killette2** [comment]
I didn't imply a literal puzzle where there's only one answer, more of a trials thing. You have to do certain task in order to advance that doesn't involve entering a code or a single solution.

Also you forgot staff of change can also make you unathi, skrell, and if you really want it, catbeast, which is normally a surgery limitation (that requires multiple people in the first place). Also as mentioned above, not only muton, slimes and xenomorphs, but also horror form.

Or fuck it, maybe we can toss a wish granter at the end of a complex away mission that way the first one to solve it gets it, and anyone else is SOL.

**Comment by icantthinkofanameritenow** [comment]
@Killette2 
The wish granter thing actually has precedent in the "wildwest.dmm" away mission, so it would work out much better than something that may or may not turn you into an antagonist (since the wish granter is guaranteed to turn you into an antag).

Someone's going to have to make its effects stronger though, right now anyone changed by it is still killable very easily if you know what you're doing.

**Comment by Killette2** [comment]
Wishgranter isn't antag if you wish for money

**Comment by icantthinkofanameritenow** [comment]
> implying anyone will wish for money

And what would they even spend it on?

**Comment by Killette2** [comment]
It's not called wish for money I don't think, something like wish for riches and it gives you a ton of smelted ores even the valuable ones

**Comment by icantthinkofanameritenow** [comment]
@Killette2 
I know that; I also question why anyone would even do that when maybe only a few jobs can actually do anything with said ore.

**Comment by Killette2** [comment]
They might do it because they're retarded. I've seen people wish for riches from wishgranted multiple times, but obviously power gets more wishes.

**Comment by MercuryDoll** [comment]
Okay, here's a dumb idea while you guys try to think up away missions. You know how we have random vaults in space right? GPS? How about we make the gateway able to spawn a portal similar to the hand-teleporter for an undetermined time and number of uses. If a brave (or foolish) soul managed to find a vault and was clever enough to bring a GPS, Telescience can make the Gateway lock onto these coordinates for a permanent portal, so the station can send a striketeam to clear the vault and rescue fallen crew. The only drawback is that the gateway will draw shittons of power while activated, enough to drain APCs with regular engine setups. This way engineers are encouraged to make high-yield engines.

**Comment by Killette2** [comment]
Couldn't you do the exact same thing with a tracking beacon and hand teles/teleporters instead?

**Comment by Exxion** [comment]
Yes, you very easily could

**Comment by Angelite191** [comment]
Ideas from the thread;
"
>Gateway jumpsuits
>Gateway backpacks with gimped storage space to promote co-operation/loot share
>No sillicons allowed?
>(Hopefully) Doesn't make away missions an antag body dump
"

"Maybe if spacemen had 'gateway' suits you had to wear to use the gateway? That way, difficulty balance could be easier too because you'd know exactly how people would start (jumpsuit only), it'd prevent people who had important roles like the captain from being able to just ditch everything and jump through, and stops people taking traitor objectives through into the away missions"

If balance are what you're worried about here then these problems are fixed by these ideas that are in no way my own.

**Comment by 9600bauds** [comment]
this werks now
